### ISSUE TITLE
Add WebSocket API

### ISSUE THREAD

**Issue by kyranet** [denoland/deno#1254]
WebSockets API (MDN): https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API
WebSocket (MDN): https://developer.mozilla.org/en-US/docs/Web/API/WebSocket
WHATWG: https://html.spec.whatwg.org/multipage/web-sockets.html#the-websocket-interface

> **Note**: The first link has a brief list of WebSocket libraries, some in JavaScript, which we can port, and some in C++.

Depends on #726

**Comment by KeKsBoTer** [comment]
Is anybody working on this?

**Comment by kitsonk** [comment]
Not right now... it really needs for the #726 to be delivered before it makes a lot of sense to try this.

**Comment by ry** [comment]
The plan is to implement Web Sockets in TS, not natively. So this would be something that would live in deno_std. Therefore closing this in favor of https://github.com/denoland/deno_std/issues/29

(I am open to implementing it natively, but we should first prove that a TS implementation is too slow. That will be many months away, if ever.)
